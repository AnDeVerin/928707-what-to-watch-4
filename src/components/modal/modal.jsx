import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ActionCreator } from '../../reducer/app/app.js';
import { getModalState, getErrorText } from '../../reducer/app/selectors.js';

const Modal = ({ text, isVisible, onClick }) => {
  const hiddenClass = isVisible ? '' : 'visually-hidden';

  return (
    <section className={`modal ${hiddenClass}`}>
      <div className="modal__content">
        <h2 className="modal__heading">Something went wrong!</h2>
        <p className="modal__text">{text}</p>

        <button className="modal__button catalog__button" onClick={onClick}>
          OK
        </button>
      </div>
    </section>
  );
};

const mapStateToProps = (state) => ({
  isVisible: getModalState(state),
  text: getErrorText(state),
});

const mapDispatchToProps = (dispatch) => ({
  onClick() {
    dispatch(ActionCreator.hideModal());
  },
});

Modal.propTypes = {
  text: PropTypes.string.isRequired,
  isVisible: PropTypes.bool.isRequired,
  onClick: PropTypes.func.isRequired,
};

export { Modal };
export default connect(mapStateToProps, mapDispatchToProps)(Modal);
