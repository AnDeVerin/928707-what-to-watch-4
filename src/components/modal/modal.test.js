import React from 'react';
import renderer from 'react-test-renderer';
import { Modal } from './modal.jsx';

describe('Modal component', () => {
  it('renders correctly', () => {
    const component = renderer
      .create(
        <Modal text={`Some error`} isVisible={true} onClick={jest.fn()} />
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
