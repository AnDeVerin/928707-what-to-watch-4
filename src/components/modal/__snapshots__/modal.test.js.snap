// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders correctly 1`] = `
<section
  className="modal "
>
  <div
    className="modal__content"
  >
    <h2
      className="modal__heading"
    >
      Something went wrong!
    </h2>
    <p
      className="modal__text"
    >
      Some error
    </p>
    <button
      className="modal__button catalog__button"
      onClick={[MockFunction]}
    >
      OK
    </button>
  </div>
</section>
`;
